core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -454,11 +466,19 @@ private void addConnectorForListener(HttpConfiguration httpConfiguration,
         SslConnectionFactory sslConnectionFactory = new SslConnectionFactory(sslContextFactory,
                 alpnConnectionFactory.getProtocol());
 
-        connector = new NetworkTrafficServerConnector(this, null, null, null, 0, 0,
-                sslConnectionFactory, alpnConnectionFactory, h2ConnectionFactory,
-                httpConnectionFactory);
+        if (proxyProtocolEnabled) {
+          connectionFactories.add(new ProxyConnectionFactory(sslConnectionFactory.getProtocol()));
+        }
+
+        connectionFactories.add(sslConnectionFactory);
+        connectionFactories.add(alpnConnectionFactory);
+        connectionFactories.add(h2ConnectionFactory);
+        connectionFactories.add(httpConnectionFactory);
       }
 
+      connector = new NetworkTrafficServerConnector(this, null, null, null, 0, 0,
+              connectionFactories.toArray(new ConnectionFactory[0]));
+
```

Comment:
nit: I think this and the similar expression at the end of the non-HTTP2 path can be combined by being moved out of the HTTP2 if-else.

Reply:
I originally didn't do this because there's one additional thing that has to happen in the HTTP2 case: https://github.com/confluentinc/rest-utils/blob/master/core/src/main/java/io/confluent/rest/ApplicationServer.java#L470. I've made this change now, and I think it's fine, but I could go either way.